src/trace_reader.rs

```diff
@@ -31,6 +31,7 @@ pub struct Sample {
     pub location: Location,
     pub sample_type: SampleType,
     pub flat_resources: ExecutionResources,
+    pub onchain_data: OnchainData,
```

Comment:
Not sure if this is a good name, this field does not store `onchain_data`. `l1_resources` maybe?